### ISSUE TITLE
Add syntax highlighting

### ISSUE THREAD

**Issue by Otiel** [ForkIssues/TrackerWin#56]
I'm surprised this hasn't been asked before. Is syntax highlighting coming? 😃

**Comment by DarkDeny** [comment]
Please make this option disableable, not everyone of us want to have this =)

**Comment by egil** [comment]
Yes please add syntax higlighting!

**Comment by xprzemekw** [comment]
That would be very much welcome feature :)

**Comment by shivan** [comment]
@TanyaPristupova are you even thinking about maybe adding syntax highlighting - is it anywhere on your roadmap? Or is it currently totally out of scope?

**Comment by egil** [comment]
Yes, please consider adding this, ideally with an easily accessible button to toggle it on off. as well as a way to specify the language, in case Fork picks the wrong highlighter.

I suggest leveraging the syntax highlighting standard used by VSCode and other editors, so that you can lean on others to provide the actual highlighting descriptions.

**Comment by Zed-Freejack** [comment]
I'd be happy too, seeing syntax highlighting in one of the next releases. Makes reviewing code changes much easier, IMHO.

**Comment by jr-hoh** [comment]
Same here, we really would appreciate basic syntax highlighting :)

**Comment by DanPristupov** [comment]
Thank you for reminding us. We implemented syntax highlighting in Fork 2.1. Please check for updates.

**Comment by jnm2** [comment]
Awesome news! Readability is so much better! 🍾

**Comment by jnm2** [comment]
After some usage, it stood out that C# highlighter seems to only understand C# as of 12 years ago at language version 5. It gets confused with `out var` which is a v6 feature.

Would you consider using the TextMate definition at <https://github.com/dotnet/csharp-tmLanguage>? It's being kept more up to date by the C# team. This definition includes support for things like [raw string literals](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/tokens/raw-string), a feature released a couple years ago. That's a language feature with a huge impact on syntax highlighting.

General TextMate format support would open the door to plug-and-play support for other languages, too. Files not highlighted currently:
- XML (e.g. `.aspx`, `.config`, `.csproj`, `.props`, ...)
- HTML
- SQL
- Markdown
- JSON
- YAML
- PowerShell
- INI (e.g. `.editorconfig`)

The TextMate definition is what VS Code uses, as well: <https://github.com/microsoft/vscode/blob/main/extensions/csharp/syntaxes/csharp.tmLanguage.json>

**Comment by Zed-Freejack** [comment]
Thanks for that. Diffs (in C++) are much more readable now :-D

**Comment by xprzemekw** [comment]
Thank very much for this! ❤️

**Comment by jr-hoh** [comment]
Thank you very much for the quick response! I really appreciate the new syntax highlighting.

What do you think about the suggestion of @jnm2? I think this would add even more to the readability of code and diffs.

**Comment by DanPristupov** [comment]
TextMate grammars work for a text editor, but don't fit well for diff because diff contains **incomplete** hunks of code.
Another issue is that TextMate grammars are based on regex lookaheads, which makes them very slow.

**Comment by jnm2** [comment]
@DanPristupov Thanks for responding! The C# member I'm talking to, who's familiar with TextMate and how VS Code uses the C# grammar, shared this with me:

> Textmate grammars work great for random hunks of code, actually. Also, yeah, they can be slow. Not much you can do about that other than having a fast textmate processor. Vscode uses the textmate grammar in several places, such as in C# markdown blocks (which are very often random hunks of code) and in cs files when you're not using semantic highlighting.

Someone else clarified for me too that TextMate is a per-line classification, with no concept of surrounding context. That lends weight to the point that it's ideal for incomplete hunks of code.

**Comment by egil** [comment]
I have noticed that syntax highlighting doesn't work for files with multiple dots (`.`) in them, e.g. `Page.razor.cs`. It does not highlight as a c# file.

**Comment by SetupCoding** [comment]
I Also seem to have the issue that @egil was describing.

Version 2.68.1 on Mac

Files like `ProductCardList.module.scss` don't have syntax highlighting at all.
Could you please doublecheck @DanPristupov  ? 
Many many thanks in advance!
